src/phoenix/datasets/experiments.py

```diff
@@ -170,3 +214,133 @@ async def async_run_experiment(example: ExampleProtocol) -> ExperimentPayload:
+
+
+def evaluate_experiment(
+    experiment: Experiment,
+    evaluator: ExperimentEvaluator,
+    name: Optional[str] = None,
+    label: Optional[str] = None,
+) -> None:
+    # define wrapper classes to coerce JSON payloads to conform to the input protocols until
+    # we flesh out our clientside models
+
+    @dataclass
+    class ExampleWrapper:
+        id: str
+        input: JSONSerializable
+        output: JSONSerializable
+
+    @dataclass
+    class RunWrapper:
+        id: str
+        output: JSONSerializable
+
+    client = _phoenix_client()
+
+    experiment_id = experiment.id
+    dataset_id = experiment.dataset_id
+    dataset_version_id = experiment.dataset_version_id
+
+    dataset_examples = client.get(
+        f"/v1/datasets/{dataset_id}/examples",
+        params={"version-id": str(dataset_version_id)},
+    ).json()
+
+    experiment_runs = client.get(f"/v1/experiments/{experiment_id}/runs").json()
+
+    # not all dataset examples have associated experiment runs, so we need to pair them up
+    # based on the example id, this assumes that the examples and experiment runs are sorted by id
+    example_run_pairs = []
+    example_cursor, run_cursor = 0, 0
+    while example_cursor < len(dataset_examples) and run_cursor < len(experiment_runs):
+        example = dataset_examples[example_cursor]
+        run = experiment_runs[run_cursor]
+        if example["id"] == run["dataset_example_id"]:
+            wrapped_example = ExampleWrapper(
+                id=example["id"], input=example["input"], output=example["output"]
+            )
+            wrapped_run = RunWrapper(id=run["id"], output=run["output"])
+            example_run_pairs.append((wrapped_example, wrapped_run))
+            example_cursor += 1
+            run_cursor += 1
+        elif example["id"] < run["dataset_example_id"]:
+            example_cursor += 1
+        else:
+            run_cursor += 1
+
+    def sync_evaluate_run(example_run: Tuple[ExampleProtocol, RunProtocol]) -> EvaluatorPayload:
+        example, run = example_run
+        start_time = datetime.now()
+        error: Optional[Exception] = None
+        try:
+            if asyncio.iscoroutinefunction(evaluator):
+                raise RuntimeError("Task is async but running in sync context")
+            else:
+                output = evaluator(input=example.input, reference=example.output, output=run.output)
+        except Exception as exc:
+            error = exc
+        finally:
+            end_time = datetime.now()
+
+        evaluator_payload = EvaluatorPayload(
+            experiment_run_id=run.id,
+            name=name if name is not None else str(evaluator),
+            annotator_kind="basic",
+            label=label if label is not None else None,
+            score=getattr(output, "score", None),
+            explanation=getattr(output, "explanation", None),
+            error=repr(error) if error else None,
+            metadata=getattr(output, "metadata", None),
```

Comment:
Not clear why we're using `getattr` here? These are fields on `EvaluationProtocol`.

Reply:
just where I felt like I could put a default, since the protocols might not be long-lived it didn't feel right to put a default there

Reply:
Feels a bit misleading since the only time the attributes are not present is when the user has incorrectly implemented their evaluator. But we probably want to error in that case to help the user fix the issue.

Reply:
I think you're right here, I'll remove the getattrs except for the annotator kind, I'm not positive about that interface just yet